sdk/core/azure-core-opentelemetry/src/opentelemetry.cpp

```diff
@@ -0,0 +1,179 @@
+      // of Azure::Core::Tracing, the Attributes passed in will be an OpenTelemetryAttributeSet
+      Azure::Core::Tracing::OpenTelemetry::OpenTelemetryAttributeSet* attributes
+          = static_cast<Azure::Core::Tracing::OpenTelemetry::OpenTelemetryAttributeSet*>(
+              options.Attributes.get());
+      newSpan = m_tracer->StartSpan(spanName, *attributes, spanOptions);
+    }
+    else
+    {
+      newSpan = m_tracer->StartSpan(spanName, spanOptions);
+    }
+
+    return std::make_shared<Azure::Core::Tracing::OpenTelemetry::OpenTelemetrySpan>(newSpan);
+  }
+
+  OpenTelemetrySpan::~OpenTelemetrySpan() {}
+
+  OpenTelemetrySpan::OpenTelemetrySpan(
+      opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> span)
+      : m_span(span)
+  {
+  }
+
+  void OpenTelemetrySpan::End(Azure::Nullable<Azure::DateTime> endTime)
+  {
+    opentelemetry::trace::EndSpanOptions options;
+    if (endTime)
+    {
+      options.end_steady_time = opentelemetry::common::SteadyTimestamp(
+          std::chrono::steady_clock::time_point(endTime.Value().time_since_epoch()));
+    }
+    m_span->End(options);
+  }
+
+  /**
+   * @brief Add the set of attributes provided to the current span.
+   */
+  void OpenTelemetrySpan::AddAttributes(AttributeSet const& attributesToAdd)
+  {
+    // Note: We make a huge assumption here: That if you're calling into the OpenTelemetry version
+    // of Azure::Core::Tracing, the Attributes passed in will be an OpenTelemetryAttributeSet
+    Azure::Core::Tracing::OpenTelemetry::OpenTelemetryAttributeSet const& attributes
+        = static_cast<Azure::Core::Tracing::OpenTelemetry::OpenTelemetryAttributeSet const&>(
+            attributesToAdd);
+    attributes.ForEachKeyValue(
+        [this](
+            opentelemetry::nostd::string_view name, opentelemetry::common::AttributeValue value) {
+          m_span->SetAttribute(name, value);
+          return true;
+        });
+  }
+
+  /**
+   * Add an Event to the span. An event is identified by a name and an optional set of attributes
+   * associated with the event.
+   */
+  void OpenTelemetrySpan::AddEvent(
+      std::string const& eventName,
+      AttributeSet const& eventAttributes)
+  {
+    Azure::Core::Tracing::OpenTelemetry::OpenTelemetryAttributeSet const& attributes
+        = static_cast<Azure::Core::Tracing::OpenTelemetry::OpenTelemetryAttributeSet const&>(
+            eventAttributes);
+
+    m_span->AddEvent(eventName, attributes);
+  }
+
+  void OpenTelemetrySpan::AddEvent(std::string const& eventName) { m_span->AddEvent(eventName); }
+
+  void OpenTelemetrySpan::AddEvent(std::exception const& ex) { m_span->AddEvent(ex.what()); }
+
+  void OpenTelemetrySpan::SetStatus(SpanStatus const& status, std::string const& statusMessage)
+  {
+    opentelemetry::trace::StatusCode statusCode = opentelemetry::trace::StatusCode::kUnset;
+    if (status == Azure::Core::Tracing::SpanStatus::Error)
```

Comment:
switch maybe ?

Reply:
C++ doesn't support using "switch" on strings :(. It would be nice if it did, but that's not an option :(.